### ISSUE TITLE
setting setting multiple scopes prevents the authentication to finish

### ISSUE THREAD

**Issue by paillave** [authts/oidc-client-ts#542]
# Problem

I authenticate with pkce using the the following settings:

```ts
const oidcSettings: UserManagerSettings = {
    authority: "https://localhost:56954/",
    client_id: "simpleIdServerWebsite",
    client_secret: "simpleIdServerWebsiteSecret",
    automaticSilentRenew: true,
    scope: "openid profile",
    response_type: "code",
    prompt: "coucou",
    loadUserInfo: true,
    redirect_uri: "https://localhost:44416",
}
```

Everything seems to be going fine except that the library doesn't seem to detect a succesfull authentication.

The following is not this interesting except for [this conclusion](#Conclusion)

# First Investigation

When looking at network activity, it appears that everything goes fine as the accessToken is requested back and returned accurately.

I activated logs in the library and see that the last news I get from it is `ResponseValidator:code processed`

So it seems that something wrong happens here:
https://github.com/authts/oidc-client-ts/blob/54fda0f58c65d12bf3f25d652eb38fea81d5063f/src/ResponseValidator.ts#L65-L67

# Debugging

Debugging your api confirms that the method `ResponseValidator.validateSigninResponse` receives a very good response with the right information (I checked with jwt.io). 

So far so good 👍 

💥 The debug fails and ends at the check if the requested authentication is openId by accessing the value of `isOpenId`
https://github.com/authts/oidc-client-ts/blob/54fda0f58c65d12bf3f25d652eb38fea81d5063f/src/ResponseValidator.ts#L65

Behind the scenes, in this situation, `isOpenId` is not an attribute but the following property:
https://github.com/authts/oidc-client-ts/blob/54fda0f58c65d12bf3f25d652eb38fea81d5063f/src/SigninResponse.ts#L74-L76

The problem is this one:
the `this.scope` is not empty, so the `split` on it is called. But... it is already is an array! 
![image](https://user-images.githubusercontent.com/4611176/170008931-0571a55e-198c-4752-a688-84a1da30d34d.png)

🔍 Let's investigate further...

As everything is strongly typed in this project, of course typescript would not transpile if you assigned an array into a string. Then, the issue must come from the `Object.assign` (as this method doesn't make any type check behind-the-scenes) on this line of `_processCode`:
https://github.com/authts/oidc-client-ts/blob/54fda0f58c65d12bf3f25d652eb38fea81d5063f/src/ResponseValidator.ts#L228

When debugging `_processCode`:
https://github.com/authts/oidc-client-ts/blob/54fda0f58c65d12bf3f25d652eb38fea81d5063f/src/ResponseValidator.ts#L216-L232

By setting a breakpoint before this line:
https://github.com/authts/oidc-client-ts/blob/54fda0f58c65d12bf3f25d652eb38fea81d5063f/src/ResponseValidator.ts#L228

The following reveals:
![image](https://user-images.githubusercontent.com/4611176/170018598-1e1df7e9-f504-4571-93e3-8c55d1f8d7a0.png)

No further process is done when getting the response to obtain the token:
https://github.com/authts/oidc-client-ts/blob/54fda0f58c65d12bf3f25d652eb38fea81d5063f/src/TokenClient.ts#L103

# Conclusion

(FYI, I wrote this ticket at the same time I was investigating, this is why maybe only what follows will be usefull to you 🤣 )

It appears that the authentication server I use (https://github.com/simpleidserver/SimpleIdServer) actually returns an array:
![image](https://user-images.githubusercontent.com/4611176/170020935-31cc741e-4b47-470a-9d10-14a8c42ba1f3.png)

# Question

Is it a bug at you side that you don't take in consideration the fact that the authentication server may return the scope by gicing an array.
Or shall I submit an an issue to the author of SimpleIdServer for not following the standard about this response?

**Comment by kherock** [comment]
Thanks for the detailed write up!

The array in the response is not typical, though I can't seem to find any wording in rfc6749 that forbids it in the response—the documentation for `scope` in the response is identical to `scope` in the authorize request. It's valid to pass multiple `scope` parameters in the request (e.g. `authorize?scope=openid&scope=profile`), so it doesn't seem unreasonable that a multivalued scope in the response could be allowed. It seems like the OAuth extensions rfc7662 and rfc8693 do explicitly define that the value in the request's JWT payload is a JSON string, although this shouldn't have much bearing over the access token response format.

Given the ambiguity in the spec, I'd be supportive of a PR here for handling the array response!

**Comment by pamapa** [comment]
The guys of SimpleIdServer agreed that it is a bug (https://github.com/simpleidserver/SimpleIdServer/issues/271) on their side. They are going to fix that, so no need to do something in this library....

**Comment by paillave** [comment]
> The guys of SimpleIdServer agreed that it is a bug ([simpleidserver/SimpleIdServer#271](https://github.com/simpleidserver/SimpleIdServer/issues/271)) on their side. They are going to fix that, so no need to do something in this library....

Indeed, I saw. For now, I'll integrate a copy of your library in my app with the fix @kherock mentionned till the new version of SimpleIdServer is on nuget.
